src/Model/Scope/CompoundCondition.php

```diff
@@ -0,0 +1,199 @@
+                // array of OR nested conditions
+                if (count($nestedCondition) === 1 && isset($nestedCondition[0]) && is_array($nestedCondition[0])) {
+                    $nestedCondition = new self($nestedCondition[0], self::OR);
+                } else {
+                    $nestedCondition = new BasicCondition(...$nestedCondition);
+                }
+            }
+
+            if ($nestedCondition->isEmpty()) {
+                continue;
+            }
+
+            $this->add(clone $nestedCondition);
+        }
+    }
+
+    public function __clone()
+    {
+        foreach ($this->elements as $k => $nestedCondition) {
+            $this->elements[$k] = clone $nestedCondition;
+            $this->elements[$k]->owner = $this;
+        }
+    }
+
+    /**
+     * Return array of nested conditions.
+     *
+     * @return AbstractCondition[]
+     */
+    public function getNestedConditions()
+    {
+        return $this->elements;
+    }
+
+    public function onChangeModel(): void
+    {
+        foreach ($this->elements as $nestedCondition) {
+            $nestedCondition->onChangeModel();
+        }
+    }
+
+    public function isEmpty(): bool
+    {
+        return empty($this->elements);
+    }
+
+    public function isCompound(): bool
+    {
+        return count($this->elements) > 1;
+    }
+
+    /**
+     * @return self::AND|self::OR
+     */
+    public function getJunction(): string
+    {
+        return $this->junction;
+    }
+
+    /**
+     * Checks if junction is OR.
+     */
+    public function isOr(): bool
+    {
+        return $this->junction === self::OR;
+    }
+
+    /**
+     * Checks if junction is AND.
+     */
+    public function isAnd(): bool
+    {
+        return $this->junction === self::AND;
+    }
+
+    /**
+     * Clears the group from nested conditions.
+     *
+     * @return static
+     */
+    public function clear()
+    {
+        foreach ($this->elements as $nestedCondition) {
+            $nestedCondition->destroy();
+        }
+
+        return $this;
+    }
+
+    public function simplify()
+    {
+        if (count($this->elements) != 1) {
+            return $this;
+        }
+
+        /** @var AbstractCondition $component */
+        $component = reset($this->elements);
+
+        return $component->simplify();
+    }
+
+    /**
+     * Use De Morgan's laws to negate.
+     *
+     * @return static
+     */
+    public function negate()
+    {
+        $this->junction = $this->junction == self::OR ? self::AND : self::OR;
+
+        foreach ($this->elements as $nestedCondition) {
+            $nestedCondition->negate();
+        }
+
+        return $this;
+    }
+
+    public function toWords(bool $asHtml = false): string
+    {
+        $parts = [];
+        foreach ($this->elements as $nestedCondition) {
+            $words = $nestedCondition->toWords($asHtml);
+
+            $parts[] = $this->isCompound() && $nestedCondition->isCompound() ? "({$words})" : $words;
```

Comment:
'(' . $words . ')'

Reply:
why?

Reply:
we should use single quoted strings only, easier to read what is string and what is variable

Reply:
done